### PR TITLE
[SQL-Gen] Add simple delete statement gen

### PR DESCRIPTION
**Pull request by willjvsmith**
* Add simple delete statement generation
  * Delete everything from table: `DELETE FROM Users;`
  * Delete based on condition: `DELETE FROM Users WHERE User.id = 123456;`
* Add basic unit tests

While adding this I was looking at what had changed on Friday, currently the conditions AST has Strings as their leaf node types. Do we want to add some semantics to restrict this?

`case class Comparison(left: String, comparison: ComparisonOperator, right: String) extends Condition`

In which case, which conditions do we want to support?

### CODE REVIEW

File: src/test/scala/temple/generate/database/package.scala [hunk 1]
```diff
@@ -170,5 +170,22 @@ package object database {
...
         |VALUES ($1, $2, $3, $4, $5, $6, $7);""".stripMargin
+
+    val deleteStatement = Delete(
+      "Users"
+    )
+
+    val postgresDeleteString: String =
+      """DELETE FROM Users;"""
+
+    val deleteStatementWithWhere = Delete(
```
Comment (Lewiky) [inline src/test/scala/temple/generate/database/package.scala:181]: Are you gonna write a test to do the DELETE with WHERE checking?

**Review by Lewiky — CHANGES_REQUESTED** [review]

File: src/main/scala/temple/generate/database/PostgresGenerator.scala [hunk 2]
```diff
@@ -74,16 +74,22 @@ object PostgresGenerator extends DatabaseGenerator {
...
         s"CREATE TABLE $tableName (\n$stringColumns\n);"
-      case Read(tableName, columns, condition) =>
+      case Read(tableName, columns, conditions) =>
         val stringColumns = columns.map(_.name).mkString(", ")
-        val stringCondition = condition.map(generateCondition) match {
-          case Some(conditionString) => List(s"WHERE $conditionString")
-          case None                  => List()
+        val stringConditions = conditions.map(generateCondition) match {
+          case Some(conditionsString) => List(s"WHERE $conditionsString")
+          case None                   => List()
```
Comment (jaylees14) [inline src/main/scala/temple/generate/database/PostgresGenerator.scala:81]: This is fine, but in the future try and avoid making changes to code that doesn't directly relate to the PR

**Review by jaylees14 — APPROVED** [review]
Looks good, just worth adding the tests as @Lewiky suggested.

With respect to your comment about making the type more rigid - this could be possible, but we'd probably end up wrapping lots of separate types in an ADT. Not sure we'd actually gain anything from doing that?

File: src/test/scala/temple/generate/database/package.scala (hunk 1 above)
Comment (c-harding) [reply src/test/scala/temple/generate/database/package.scala:181]: Will accept when complete

**Review by c-harding — CHANGES_REQUESTED** [review]

**Review by c-harding — APPROVED** [review]

File: src/main/scala/temple/generate/database/PostgresGenerator.scala (hunk 2 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/database/PostgresGenerator.scala:81]: I know what you mean, but this does kinda relate to the PR. Had to write the same stuff for the Delete case and noticed it was wrong.

**Review by Lewiky — APPROVED** [review]

**Pull request merged**